src/audio/module_adapter/module/waves/waves.c

```diff
@@ -659,6 +717,22 @@ static int waves_codec_init(struct processing_module *mod)
 	return ret;
 }
 
+/* apply config blob */
+static int waves_effect_apply_config_blob_cache(struct processing_module *mod)
```

Comment:
I beg to state that the commit complicates the design. After that there will be both waves_effect_apply_config() and waves_effect_apply_config_cache(). Moreover, waves_effect_config(), waves_effect_cache_config(), and waves_effect_message().

(Please correct me if there is devil in the details.) I do believe the logic is fair enough as I mentioned in the former comment. That is, two reaction chain in total which can be triggered from external:
1. set config blob from external:
```
(...) -> waves_codec_set_configuration() -> waves_effect_apply_config() -> waves_effect_config_save_to_cache()
```
2. start module (trigger prepare) from external: 
```
(...) -> waves_codec_prepare() -> waves_effect_config_load_from_cache()
```
then handle what if...
   - cache is empty when calling `waves_effect_config_load_from_cache()`?
   - module is processing when calling `waves_effect_config_save_to_cache()`?